src/main/java/com/google/api/generator/engine/ast/JavaDocComment.java

```diff
@@ -0,0 +1,105 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.api.generator.engine.ast;
+
+import com.google.auto.value.AutoValue;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+@AutoValue
+public abstract class JavaDocComment implements Comment {
+
+  @Nullable
+  public abstract String deprecated();
+
+  public abstract ImmutableList<String> comments();
+
+  public static Builder builder() {
+    return new AutoValue_JavaDocComment.Builder().setDeprecated("");
+  }
+
+  @AutoValue.Builder
+  public abstract static class Builder {
+    protected abstract ImmutableList.Builder<String> commentsBuilder();
+
+    public abstract Builder setDeprecated(String deprecatedText);
+
+    public Builder addComment(String comment) {
+      commentsBuilder().add(comment);
+      return this;
+    }
+
+    public Builder addParam(String name, String description) {
+      String parameter = String.format("%s %s %s", "@param", name, description);
+      commentsBuilder().add(parameter);
+      return this;
+    }
+
+    public Builder addThrowsText(String type, String description) {
```

Comment:
Do we usually handle more than one `@thorws`? If not, can this be `setThrows(type, description)` instead?

Reply:
There's only one `@throw` in all cases, but technically according to doc `Builder methods must have 0 or 1 parameters`. We might need this `addThrowsText` to take 2 args for us. Thanks! Please let me know if there's a better way to do this.

Reply:
This should still be `setThrows(type, description)` since there will only be one of them, and as we've seen in this class, `Builder` methods can take two arguments.

Also, I noticed that the corresponding getter is missing. For consistency, either add back a getter (under a misspelled name like `throes`) or remove the `deprecated()` method if these getters are not needed.

Reply:
Oh, I got what you mean, thanks for the explanation.
Removed the getter of `deprecated` `throwsText` and just have `setDeprecated` & `setThrows` builder.